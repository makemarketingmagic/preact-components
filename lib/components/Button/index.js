'use strict';

exports.__esModule = true;
exports.default = undefined;

var _preact = require('preact');

var _Button = require('./Button.less');

var _Button2 = _interopRequireDefault(_Button);

var _material = require('material-design-lite/material');

var _material2 = _interopRequireDefault(_material);

var _MUIComponents = require('../common/MUIComponents');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }
// eslint-disable-next-line no-unused-vars


var Button = function (_Component) {
    _inherits(Button, _Component);

    function Button() {
        _classCallCheck(this, Button);

        return _possibleConstructorReturn(this, _Component.apply(this, arguments));
    }

    Button.prototype.render = function render() {
        var _props = this.props,
            children = _props.children,
            _props$Icon = _props.Icon,
            Icon = _props$Icon === undefined ? null : _props$Icon,
            _props$iconLeft = _props.iconLeft,
            iconLeft = _props$iconLeft === undefined ? false : _props$iconLeft,
            _props$iconRight = _props.iconRight,
            iconRight = _props$iconRight === undefined ? false : _props$iconRight,
            _props$secondary = _props.secondary,
            secondary = _props$secondary === undefined ? false : _props$secondary,
            _props$disabled = _props.disabled,
            disabled = _props$disabled === undefined ? false : _props$disabled;

        return (0, _preact.h)(
            _MUIComponents.MUIButton,
            {
                raised: true,
                primary: secondary,
                colored: !secondary,
                accent: !secondary,
                disabled: disabled
            },
            (0, _preact.h)(
                'div',
                { 'class': _Button2.default.buttonContent },
                iconLeft && Icon && (0, _preact.h)(Icon, { color: 'inherit' }),
                (0, _preact.h)(
                    'span',
                    { style: {
                            marginLeft: iconLeft ? 8 : 0,
                            marginRight: iconRight ? 8 : 0
                        } },
                    children
                ),
                iconRight && Icon && (0, _preact.h)(Icon, { color: 'inherit' })
            )
        );
    };

    return Button;
}(_preact.Component);

exports.default = Button;