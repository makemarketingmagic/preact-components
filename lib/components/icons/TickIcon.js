"use strict";

exports.__esModule = true;
exports.default = undefined;

var _preact = require("preact");

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var TickIcon = function (_Component) {
    _inherits(TickIcon, _Component);

    function TickIcon() {
        _classCallCheck(this, TickIcon);

        return _possibleConstructorReturn(this, _Component.apply(this, arguments));
    }

    TickIcon.prototype.render = function render() {
        var _props = this.props,
            width = _props.width,
            height = _props.height,
            color = _props.color;

        return (0, _preact.h)(
            "svg",
            { width: width || 14, height: height || 10 },
            (0, _preact.h)("path", { fill: "none", "fill-rule": "evenodd", stroke: color || "#FFFFFF", "stroke-linecap": "round", "stroke-linejoin": "round", "stroke-width": "2", d: "M1 5l4 4 8-8" })
        );
    };

    return TickIcon;
}(_preact.Component);

exports.default = TickIcon;