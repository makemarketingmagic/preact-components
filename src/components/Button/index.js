import { h, Component } from 'preact'
import style from './Button.less'
// eslint-disable-next-line no-unused-vars
import mdl from 'material-design-lite/material';
import { MUIButton } from '../common/MUIComponents';

export default class Button extends Component {
    render() {
        const { children, Icon = null, iconLeft = false, iconRight = false, secondary = false, disabled = false } = this.props
        return (
            <MUIButton
                raised={true}
                primary={secondary}
                colored={!secondary}
                accent={!secondary}
                disabled={disabled}
            >
                <div class={style.buttonContent}>
                    {iconLeft && Icon && <Icon color='inherit' />}
                    <span style={{
                        marginLeft: iconLeft ? 8 : 0,
                        marginRight: iconRight ? 8 : 0
                    }}>{children}</span>
                    {iconRight && Icon && <Icon color='inherit' />}
                </div>
            </MUIButton>
        )
    }
}