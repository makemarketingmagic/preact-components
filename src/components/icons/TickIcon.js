import { h, Component } from 'preact'

export default class TickIcon extends Component {
    render() {
        const { width, height, color } = this.props;
        return (
            <svg width={width || 14} height={height || 10}>
                <path fill="none" fill-rule="evenodd" stroke={color || "#FFFFFF"} stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M1 5l4 4 8-8" />
            </svg>
        )
    }
}