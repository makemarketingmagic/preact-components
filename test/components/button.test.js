import { h } from 'preact';
import render from 'preact-render-to-string';
import Button from '../../src/components/Button';
import TickIcon from './../../src/components/icons/TickIcon';

describe('Button', () => {
    it('should render', () => {
        const element = render(<Button>Hello World</Button>)
        expect(element).toMatchSnapshot()
    });

    it('should render secondary', () => {
        const element = render(<Button secondary>Hello World</Button>)
        expect(element).toMatchSnapshot()
    });

    it('should render disabled', () => {
        const element = render(<Button disabled>Hello World</Button>)
        expect(element).toMatchSnapshot()
    });

    it('should render with Icon on the left', () => {
        const element = render(<Button Icon={TickIcon} iconLeft>Hello World</Button>)
        expect(element).toMatchSnapshot()
    })

    it('should render with Icon on the right', () => {
        const element = render(<Button Icon={TickIcon} iconRight>Hello World</Button>)
        expect(element).toMatchSnapshot()
    })
})