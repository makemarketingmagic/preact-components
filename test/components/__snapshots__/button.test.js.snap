// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<button raised colored accent class="mdl-button mdl-js-ripple-effect mdl-js-button mdl-button--raised mdl-button--colored mdl-button--accent">
  <div class="buttonContent">
    <span style="margin-left: 0px; margin-right: 0px;">Hello World</span>
  </div>
</button>
`;

exports[`Button should render disabled 1`] = `
<button raised colored accent disabled class="mdl-button mdl-js-ripple-effect mdl-js-button mdl-button--raised mdl-button--colored mdl-button--accent mdl-button--disabled">
  <div class="buttonContent">
    <span style="margin-left: 0px; margin-right: 0px;">Hello World</span>
  </div>
</button>
`;

exports[`Button should render secondary 1`] = `
<button raised primary class="mdl-button mdl-js-ripple-effect mdl-js-button mdl-button--raised mdl-button--primary">
  <div class="buttonContent">
    <span style="margin-left: 0px; margin-right: 0px;">Hello World</span>
  </div>
</button>
`;

exports[`Button should render with Icon on the left 1`] = `
<button raised colored accent class="mdl-button mdl-js-ripple-effect mdl-js-button mdl-button--raised mdl-button--colored mdl-button--accent">
  <div class="buttonContent">
    <svg width="14" height="10">
      <path fill="none" fill-rule="evenodd" stroke="inherit" stroke-linecap="round"
      stroke-linejoin="round" stroke-width="2" d="M1 5l4 4 8-8"></path>
    </svg>
    <span style="margin-left: 8px; margin-right: 0px;">Hello World</span>
  </div>
</button>
`;

exports[`Button should render with Icon on the right 1`] = `
<button raised colored accent class="mdl-button mdl-js-ripple-effect mdl-js-button mdl-button--raised mdl-button--colored mdl-button--accent">
  <div class="buttonContent">
    <span style="margin-left: 0px; margin-right: 8px;">Hello World</span>
    <svg width="14" height="10">
      <path fill="none" fill-rule="evenodd" stroke="inherit" stroke-linecap="round"
      stroke-linejoin="round" stroke-width="2" d="M1 5l4 4 8-8"></path>
    </svg>
  </div>
</button>
`;
